sycl/include/sycl/vector_preview.hpp

```diff
@@ -983,90 +670,152 @@ template <typename Type, int NumElements> class vec {
     }
   }
 
+  /******************* sycl::vec math operations ***********************/
+
+  template <typename T>
+  using IsByte = typename
+#if (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE != 0)
+      std::is_same<T, std::byte>;
+#else
+      std::false_type;
+#endif
+
 #ifdef __SYCL_BINOP
 #error "Undefine __SYCL_BINOP macro"
 #endif
 
-#ifdef __SYCL_USE_EXT_VECTOR_TYPE__
-#define __SYCL_BINOP(BINOP, OPASSIGN, CONVERT)                                 \
-  friend vec operator BINOP(const vec &Lhs, const vec &Rhs) {                  \
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
Please refactor into 

```
#ifdef __SYCL_DEVICE_ONLY__
#define BINOP_BASE
#else
#define BINOP_BASE
#endif
#define BINOP() \
  BINOPT_BASE \
  <delegation to base, same for host/device>
```

`BINOP_BASE` name isn't necessary, there may be a better choice for it.